### PR TITLE
Register template for keda-v0.0.1-alpha

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- add module template for keda module (v0.0.1-alpha)

How to use it:

[Get latest (unstable) binary for kyma cli](https://github.com/kyma-project/module-manager/issues/157)
```
kyma-unstable alpha deploy

kubectl apply -k modules/alpha                                                                     
moduletemplate.operator.kyma-project.io/moduletemplate-keda created
```

**Related issue(s)**
https://github.com/kyma-project/keda-manager/issues/58
https://github.com/kyma-project/module-manager/issues/157

### CODE REVIEW

File: modules/alpha/kustomization.yaml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 apiVersion: kustomize.config.k8s.io/v1alpha1
 kind: Component
 
+resources:
+- moduletemplate-keda-v0.0.1-alpha.yaml
```
Comment (jakobmoellersap) [inline modules/alpha/kustomization.yaml:5]:
```suggestion
- moduletemplate-keda.yaml
```
no need for the version

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    "operator.kyma-project.io/descriptor-schema-version": "v2"
+spec:
+  target: remote
+  channel: alpha
+  data:
+    apiVersion: operator.kyma-project.io/v1alpha1
+    kind: Keda
+    metadata:
+      labels:
+        operator.kyma-project.io/kyma-name: keda-sample
```
Comment (jakobmoellersap) [inline modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml:22]: this Kyma name is opinionated and will not be relevant in non controlled remote environments. Could you remove this?

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      logging:
+        operator:
+          level: "debug"
+  descriptor:
+    component:
+      componentReferences: []
+      name: kyma.project.io/module/keda
+      provider: internal
+      repositoryContexts:
+      - baseUrl: europe-west4-docker.pkg.dev/sap-se-cx-kyma-wookiee/dev-registry/unsigned
```
Comment (jakobmoellersap) [inline modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml:34]: this base-url looks like it is not the official Kyma dev registry. Could you change the storage location to one that is within Kyma dev?

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      - access:
+          digest: sha256:f4a599c4310b0fe9133b67b72d9b15ee96b52a1872132528c83978239b5effef
+          type: localOciBlob
+        name: config
+        relation: local
+        type: yaml
+        version: 0.0.1-alpha
+      sources:
+      - access:
+          commit: 157f9933b29f107e9030b60411ee7f0e9a3bea7d
```
Comment (jakobmoellersap) [inline modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml:54]: commit cannot be verified, please double check it exists. I will have to base my review on main (8622f7f4eab5b64994a89eec5edd0f3e1082f8d6)

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          ref: refs/heads/main
```
Comment (jakobmoellersap) [inline modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml:55]: on main, RBACs are still in `*` import. This cannot be accepted for module submission.

Additionally, your controller requests `*` on get of all resources in the cluster, this is invalid.

**Review by jakobmoellersap — CHANGES_REQUESTED** [review]

**Comment by kwiatekus** [comment]
Depends on https://github.com/kyma-project/test-infra/pull/6490

File: modules/alpha/kustomization.yaml (hunk 1 above)
Comment (kwiatekus) [reply modules/alpha/kustomization.yaml]: done

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml (hunk 2 above)
Comment (kwiatekus) [reply modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml]: done

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml (hunk 4 above)
Comment (kwiatekus) [reply modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml]: fixed

**Review by pPrecel — APPROVED** [review]

File: modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml (hunk 5 above)
Comment (m00g3n) [reply modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml]: > on main, RBACs are still in `*` import. This cannot be accepted for module submission. Specifically, your controller requests `*` on get of all resources in the cluster, this is invalid.

We are limited with keda requirements:
- https://github.com/kyma-project/keda-manager/blob/main/module-chart/templates/10-keda-clusterrole.yaml#L37

Comment (m00g3n) [reply modules/alpha/moduletemplate-keda-v0.0.1-alpha.yaml]: > On top of that, your controller has no finalizer defined in https://github.com/kyma-project/keda-manager/blob/8622f7f4eab5b64994a89eec5edd0f3e1082f8d6/controllers/keda_controller.go#L75 and will thus break on deletion of your resource by not waiting and reporting state correctly.

thx, updated in [this PR](https://github.com/kyma-project/keda-manager/commit/a0e86481d91cb126972e2cba82361c11d6e4d2db#diff-14f943bad946b1bf34063f0b9321468d99706ebc2bafec6eb129fb51ba470375R78)

**Comment by m00g3n** [comment]
switched to https://github.com/kyma-project/kyma/pull/16347

**Pull request closed without merging**